Yatima/Expr.lean

```diff
@@ -1,62 +1,86 @@
-import Yatima.Env
 import Yatima.Univ
 
 namespace Yatima
 
-  inductive Bind
-  | default
-  | implicit
-  | strictImplicit
-  | instImplicit
-  deriving BEq, Inhabited
-
-  inductive LitType
+inductive LitType
   | nat : LitType
   | str : LitType
   deriving BEq, Inhabited
 
-  inductive Literal
+inductive Literal
   | nat : Nat → Literal
   | str : String → Literal
   deriving BEq, Inhabited
 
-  inductive Expr
+inductive BinderInfo
+  | default
+  | implicit
+  | strictImplicit
+  | instImplicit
+  | auxDecl
+  deriving BEq, Inhabited
+
+inductive Expr
   | var   : Name → Nat → Expr
-  | sort  : Univ → Expr
-  | const : Name → ConstCid → List Univ → Expr
-  | app   : Expr → Expr → Expr
-  | lam   : Name → Bind → Expr → Expr → Expr
-  | pi    : Name → Bind → Expr → Expr → Expr
-  | letE  : Name → Expr → Expr → Expr → Expr
+  | sort  : UnivCid → Expr
+  | const : Name → ConstCid → List UnivCid → Expr
+  | app   : ExprCid → ExprCid → Expr
+  | lam   : Name → BinderInfo → ExprCid → ExprCid → Expr
+  | pi    : Name → BinderInfo → ExprCid → ExprCid → Expr
+  | letE  : Name → ExprCid → ExprCid → ExprCid → Expr
   | lit   : Literal → Expr
   | lty   : LitType → Expr
-  | fix   : Name → Expr → Expr
+  | fix   : Name → ExprCid → Expr
```

Comment:
These were changed too! I replaced the raw objects by their respective CID references

Reply:
Ah, only the Univ should be replaced with a UnivCid, not Expr